src/app/repository/platform-repository.service.ts

```diff
@@ -140,11 +147,17 @@ export class PlatformRepositoryService {
 
             const missing = projectIDs.filter(id => openProjects.indexOf(id) === -1);
 
-            if (missing.length) {
-                return this.patch({openProjects: openProjects.concat(missing)}).toPromise();
+            if (missing.length === 0) {
+                return Promise.resolve();
             }
 
-            return Promise.resolve();
+            this.auth.getActive().take(1).subscribe((active) => {
```

Comment:
Setting expansion should not be bound to adding a project, same as for local repository.
`PlatformRepository` also shouldn't need the `AuthService`, since every call to backend is bound to the user context there.